### DISCUSSION TITLE
Can i use NJSPC to control an Intelliflo 2 VST and MasterTemp heater, but with an override for my pool guy?

### DISCUSSION THREAD

**Discussion by dkossman** [tagyoureit/nodejs-poolController#965] [Q&A] [answered]
I have a 2017 Pentair Intelliflo 2 VST pump model 011055, circa 2017.  It is running Intelliflo firmware 1.23-VS.  

I just ordered a new MasterTemp heater to replace my 45-year old failing gas heater, along with a new Pentair filter.  I also ordered, but canceled, an Intelliconnect panel.  I had hoped to be able to use the Intelliconnect and the Pentair Home app to control my pool pump and set the temperature remotely.  However, i called Pentair tech support to double-check, and they said that Intelliconnect will NOT work with Intelliflo VST pumps older than 2019, and that there is no firmware update to remedy this.  The pump is working fine so i really don't want to replace it just for this.

So - just wanted to confirm that i can use NJSPC instead - I had this running on a Raspberry Pi (zero) a few years ago, but gave up because it was impossible for my pool maintenance person to control the pump when he came to service the pool as it disables the pump control panel.

so, to my questions (apologies if these are all answered in an FAQ somewhere):

1. I assume that in addition to the Pi and a relay board to turn on/off the heater,  i'd need a temperature probe inserted in the return line.  The probes i see for pool systems seem all to be thermistor-based, are there digital ones (DS18B20-based) or hardware fittings to mount a standard DS18B20 probe onto a pool return line?  

2. Is there a way with  NJSPC to set up manual controls to allow the pool guy to control the pump when he services the pool?  Could i install a DPDT switch to allow him to temporarily disconnect the RS485 bus from the Pi, so that the existing pump control panel can be used?  Then flip it back when he is done to let NJSPC take over again?  Or does NJSPC have some other mechanism to support the "pool guy" use case?

thanks for any advice.

**Comment by rstrouse** [comment, marked as answer]
You can create a hardwired override to standby or timeout using a hardwired pushbutton #508

**Reply by dkossman** [reply]
thanks for the quick response, - will follow the discussion that you linked.

**Reply by rstrouse** [reply]
The inputs from gpio and Sequent IO will now allow you to send toggle commands.  This means you can now have a momentary pushbutton to turn on and off a circuit from REM.

**Comment by dkossman** [comment]
On a related topic, I have an existing NodeJS server on a Pi with an I2C relay board that does the following:
- periodically collects temperatures from DS18B20 air, pool and spa probes and pushes the data into InfluxDB
- provides REST api's to 
   - open/close the pool cover, 
   - open/close the diverter valves for the spa 
   - operate the bubbler
   - turn on/off the heater (old heater had a fireman's switch)
   - I use this server to integrate with HomeKit via Homebridge, using a homebridge HTTP plugin to call my REST api's.  So we can use Siri to operate the pool equipment.  this makes my wife happy.
   
I've just replaced my heater with a Pentair MasterTemp, which has an RJ45 interface, which means i can control my pump and heater with NJSPC in a much better way, including starting/stopping service mode (as discussed above)

So to the questions -
- assuming i install NJSPC and REM (not sure about Nixie), do i still need my NJS server to provide temperature logging and a mechanism to support Homebridge/HomeKit?  
- could REM call my existing REST api's?
- I looked for the NJSPC Homebridge plugin, but it's not clear whether it still works with the current NJSPC 8.0.x as there haven't been any updates for quite a while
- Bottom line is that HomeKit support is really important for my particular household

 thanks for any advice...

**Reply by rstrouse** [reply]
> assuming i install NJSPC and REM (not sure about Nixie), do i still need my NJS server to provide temperature logging and a mechanism to support Homebridge/HomeKit?

Nixie is the pool controller that handles everything from circuits, schedules, heaters, pumps, valves... etc. without the need for another pool controller.  It will provide that interface between your water temperature and the MasterTemp over RS485.   REM, njsPC, and dashPanel all work in unison to provide all the functionality from configuration and setup to managing the hardware.  REM also has a 1-wire interface to read the DBS18B20 sensors.

> could REM call my existing REST APIs?

I am not sure that is required.  However, njsPC could call them if you need it to but the circuit control to the I2c relay may already be supported.

> I looked for the NJSPC Homebridge plugin, but it's not clear whether it still works with the current NJSPC 8.0.x as there haven't been any updates for quite a while

I do not use Homebridge but there are a few folks on the discussions that are using the gadget-monk version.

**Comment by dkossman** [comment]
thanks for the info - Seems like i should be able to retire my nodeJS server and use Nixie/REM/Dashpanel/NJSPC.   but...
- I assume that i'll need to install a separate temperature probe in the pump return line for temperature control, right?  
- Can NJSPC switch the MasterTemp between Spa and Pool modes over RS485?
- the pool cover open and close relays each need to be activated for a specific amount of time to allow the cover to open or close, then released, so that the physical key switch still works.  not clear whether REM has this capability? 
- if I set up Nixie and REM to control my valves, bubbler, etc - will these be available to HomeKit via @gadgetmonk's plugin?  I'll try posting a question in that repo unless you happen to know the answer.

thanks as usual for the support!

**Reply by rstrouse** [reply]
The answer to whether you need a 10k temp sensor is… you can use your existing 1wire implementation is you are happy with it.  However a 10k probe in the pipe is so much more straight forward.

Yes njsPC supports the RS485 interface for MasterTemp.  Further Nixie will manage the intake, return, spillway, and drain modes for the valves.  It will also allow you to have setpoints for pool and spa.  Each body and the corresponding heat modes are managed separately.  These setpoints can also be scheduled.   There is also an egg timer to ensure circuits are not left on.
